core/hyphenator-liang.lua

```diff
@@ -113,23 +113,19 @@ local hyphenateNode = function (node)
   if #segments > 1 then
     local hyphen = SILE.shaper:createNnodes(SILE.settings:get("font.hyphenchar"), node.options)
     local newnodes = {}
-    local leadingApostrophe, nextLeadingApostrophe
     for j, segment in ipairs(segments) do
+      local leadingApostrophe = nil
       if segment == "" then
         SU.dump{ j, segments }
         SU.error("No hyphenation segment should ever be empty", true)
       end
       if node.options.language == "tr" then
-        local nextSegment = segments[j+1]
-        leadingApostrophe, nextLeadingApostrophe = nextLeadingApostrophe, nextSegment and luautf8.match(nextSegment, "^['’]")
-        if j < #segments then
-          segments[j+1] = luautf8.gsub(nextSegment, "^['’]", "")
-        end
-        if leadingApostrophe then
-          local replacement = SILE.shaper:createNnodes(leadingApostrophe, node.options)
-          local newNode = SILE.nodefactory.discretionary({ prebreak = hyphen, replacement = replacement })
-          newNode.parent = node
-          table.insert(newnodes, newNode)
+        local hasNextApostrophe = j < #segments and luautf8.match(segments[j+1], "^['’]")
```

Comment:
As a general rule variables or methods named `has...` would be booleans, lets name this based on the content it may or may not have, not just whether it has it or not.

Reply:
Agreed, and your `nextApostrophe` is better indeed.